scripts/mp-2023/plot_performance_over_available_memory.py

```diff
@@ -43,18 +54,27 @@
             else:
                 type = 'mmap-based'
                 memory = float(filename.split("_")[5][:-7])
-                #calculate latency
-
+                #assuming only int memory limit sizes
+                memory_key = str(int(memory))
+                execution_pgfaults = 0
+                if memory_key in pgfault_data:
+                    pgfaults_stats = pgfault_data[memory_key]
+                    before_memory_stats = pgfaults_stats['before']['memory_stat']
+                    after_memory_stats = pgfaults_stats['after']['memory_stat']
+                    execution_pgfaults = parse_occurred_pagefaults(after_memory_stats) - parse_occurred_pagefaults(before_memory_stats)
                 #append data to data frame
-                data_df = data_df.append({"memory": float(memory), "type": type, "latency": int(latency_all)}, ignore_index=True)
+                data_df = data_df.append({"memory": memory, "type": type, "latency": int(latency_all), "pgfaults": execution_pgfaults}, ignore_index=True)
+
 
-print(data_df)
 benchmark_results = sns.lineplot(data=data_df, x="memory", y="latency", hue="type", marker='o', linestyle='--')
+benchmark_results_second = sns.lineplot(data=data_df, x="memory", y="pgfaults", color='red', marker='o', linestyle='dotted', ax=benchmark_results.twinx())
 
 benchmark_results.set(
     xlabel="Available RAM in GB", ylabel="Latency in ms/iter (Sum over all Queries)", title=f"MMAP-based Single-Threaded Hyrise\nSum of Average Latency over all Queries Depending on Memory Limitation."
```

Comment:
For title case in all titles/axis labels
```suggestion
    xlabel="Available RAM in GB", ylabel="Latency in ms/iter (Sum over all Queries)", title=f"MMAP-Based Single-Threaded Hyrise\nSum of Average Latency over all Queries Depending on Memory Limitation."
```